# Assessing
Use the space below to explore `all_alpha_08.csv` and `all_alpha_18.csv` to answer the quiz questions below.

In [1]:
import warnings #to disable warnings
warnings.filterwarnings('ignore') #disables warnings

#import numpy as np 
import pandas as pd #for data processing
import matplotlib.pyplot as plt #for plotting
import seaborn as sns #for plotting
#make plots visible inside the jupyter notebook
%matplotlib inline 
%config InlineBackend.figure_format = 'retina'

from sklearn.linear_model import LinearRegression #calc the trend-data for later plots
#import missingno as ms #visualize missing data

#improving the plot - quality (from: http://blog.rtwilson.com/how-to-get-nice-vector-graphics-in-your-exported-pdf-ipython-notebooks/)
from IPython.display import set_matplotlib_formats 
set_matplotlib_formats('png', 'pdf')

print('Import Libraries Complete')

Import Libraries Complete


In [2]:
#load the datasets
df_08 = pd.read_csv("./all_alpha_08.csv")
df_18 = pd.read_csv("./all_alpha_18.csv")

print('Load Datasets Complete')

Load Datasets Complete


*Number of samples in each dataset*

In [26]:
#df_08.shape
df_18.shape

(1611, 18)

In [75]:
df_08.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2404 entries, 0 to 2403
Data columns (total 18 columns):
Model                   2404 non-null object
Displ                   2404 non-null float64
Cyl                     2205 non-null object
Trans                   2205 non-null object
Drive                   2311 non-null object
Fuel                    2404 non-null object
Sales Area              2404 non-null object
Stnd                    2404 non-null object
Underhood ID            2404 non-null object
Veh Class               2404 non-null object
Air Pollution Score     2404 non-null object
FE Calc Appr            2205 non-null object
City MPG                2205 non-null object
Hwy MPG                 2205 non-null object
Cmb MPG                 2205 non-null object
Unadj Cmb MPG           2205 non-null float64
Greenhouse Gas Score    2205 non-null object
SmartWay                2404 non-null object
dtypes: float64(2), object(16)
memory usage: 338.1+ KB


In [62]:
df_08.head()

Model  Displ      Cyl    Trans Drive      Fuel Sales Area Stnd  \
0  ACURA MDX    3.7  (6 cyl)  Auto-S5   4WD  Gasoline         CA   U2   
1  ACURA MDX    3.7  (6 cyl)  Auto-S5   4WD  Gasoline         FA   B5   
2  ACURA RDX    2.3  (4 cyl)  Auto-S5   4WD  Gasoline         CA   U2   
3  ACURA RDX    2.3  (4 cyl)  Auto-S5   4WD  Gasoline         FA   B5   
4   ACURA RL    3.5  (6 cyl)  Auto-S5   4WD  Gasoline         CA   U2   

   Underhood ID    Veh Class Air Pollution Score FE Calc Appr City MPG  \
0  8HNXT03.7PKR          SUV                   7          Drv       15   
1  8HNXT03.7PKR          SUV                   6          Drv       15   
2  8HNXT02.3DKR          SUV                   7          Drv       17   
3  8HNXT02.3DKR          SUV                   6          Drv       17   
4  8HNXV03.5HKR  midsize car                   7          Drv       16   

  Hwy MPG Cmb MPG  Unadj Cmb MPG Greenhouse Gas Score SmartWay  
0      20      17        22.0527                    4       no  
1      20      17        22.0527                    4       no  
2      22      19        24.1745                    5       no  
3      22      19        24.1745                    5       no  
4      24      19        24.5629                    5       no

In [80]:
df_08['Fuel'].unique()
#df_18['Fuel'].unique()


array(['Gasoline', 'ethanol/gas', 'ethanol', 'diesel', 'CNG'], dtype=object)

2404 rows in df_08 - 1611 rows in df_18

Number of columns in each dataset


In [47]:
#df_08.info
df_18.columns

Index(['Model', 'Displ', 'Cyl', 'Trans', 'Drive', 'Fuel', 'Cert Region',
       'Stnd', 'Stnd Description', 'Underhood ID', 'Veh Class',
       'Air Pollution Score', 'City MPG', 'Hwy MPG', 'Cmb MPG',
       'Greenhouse Gas Score', 'SmartWay', 'Comb CO2'],
      dtype='object')

Duplicate rows in each dataset


In [28]:
#df_08.duplicated().sum()
df_18.duplicated().sum()


0

Datatypes of columns


In [41]:
#df_08.dtypes
df_18.dtypes

Model                    object
Displ                   float64
Cyl                     float64
Trans                    object
Drive                    object
Fuel                     object
Cert Region              object
Stnd                     object
Stnd Description         object
Underhood ID             object
Veh Class                object
Air Pollution Score       int64
City MPG                 object
Hwy MPG                  object
Cmb MPG                  object
Greenhouse Gas Score      int64
SmartWay                 object
Comb CO2                 object
dtype: object

Features with missing values


In [30]:
#df_08.isna().sum()
df_18.isna().sum()

Model                   0
Displ                   2
Cyl                     2
Trans                   0
Drive                   0
Fuel                    0
Cert Region             0
Stnd                    0
Stnd Description        0
Underhood ID            0
Veh Class               0
Air Pollution Score     0
City MPG                0
Hwy MPG                 0
Cmb MPG                 0
Greenhouse Gas Score    0
SmartWay                0
Comb CO2                0
dtype: int64

Number of non-null unique values for features in each dataset


In [40]:
df_08.nunique()
#df_18.nunique()

Model                   436
Displ                    47
Cyl                       8
Trans                    14
Drive                     2
Fuel                      5
Sales Area                3
Stnd                     12
Underhood ID            343
Veh Class                 9
Air Pollution Score      13
FE Calc Appr              2
City MPG                 39
Hwy MPG                  43
Cmb MPG                  38
Unadj Cmb MPG           721
Greenhouse Gas Score     20
SmartWay                  2
dtype: int64

What those unique values are and counts for each


1757